### DISCUSSION TITLE
Next.js code splitting implicitly covers up the use case of patterns like HOCs and render props?

### DISCUSSION THREAD

**Discussion by prerakhere** [vercel/next.js#38314] [Help]
The code splitting section of next.js docs (https://nextjs.org/learn/foundations/how-nextjs-works/code-splitting) says - 
"Any code shared between pages is also split into another bundle to avoid re-downloading the same code on further navigation."

Does this imply that the splitted code has the optimizations that patterns like HOCs and render props offer in terms of logic re-usability between the components?

**Comment by icyJoseph** [comment]
Sorry for answering with a question, but what do you mean

> ...has the optimizations that patterns like HOCs and render props offer in terms of logic re-usability between the components...

A pattern like a HoC, or render props, or custom hooks, is used to extend our React components with our business logic, and make it reusable through our application. Using the term `business logic` as a catch-all for state-management.

Code splitting in the other hand is all about chunking the JavaScript so that the browser doesn't have to block the rendering because it has to parse a large JavaScript file.

In Next.js code splitting is used to separate the code used by various pages, and code used in many pages is all bundles together, so that when you visit page A, and then transition to page B, you wouldn't have to download that shared code again, though you'd still need to download the code for page B. Next.js does prefetching of this code for you.

Do remember that Next.js offers a SPA-like experience, albeit with pages that are capable of working as stand alone entry points, once you are at page A, you can transition without reloading to other pages.

**Comment by prerakhere** [comment]
Thanks for your input on this. 

I clearly got confused and took that statement the other way. 

What I inferred from that statement is that next automatically identifies the code that is shared across pages (even if the code is written in not-so-DRY fashion) and split that into separate bundle as an optimization alternative for what patterns like HOCs and render props help us to achieve.
